components/TARGET_PSA/services/crypto/COMPONENT_SPE/psa_crypto_partition.c

```diff
@@ -65,7 +69,12 @@ static void psa_crypto_free_operation( void )
 
         case PSA_IPC_CALL:
         {
-            mbedtls_psa_crypto_free();
+            /** perform crypto_free iff the number of init-s
+             * is equal to the number of free-s
+             */
+            --psa_spm_init_refence_counter;
```

Comment:
Major: what do you think happens if a client calls `PSA_CRYPTO_FREE` without a corresponding `PSA_CRYPTO_INIT`?

I'm rating this ”major“ and not “critical” because it's a critical security issue, but this version is for evaluation only, so it's ok to release with it if we document it as a known critical security issue.